plan/logical_plan_builder.go

```diff
@@ -63,14 +63,24 @@ func (p *LogicalAggregation) collectGroupByColumns() {
 func (b *planBuilder) buildAggregation(p LogicalPlan, aggFuncList []*ast.AggregateFuncExpr, gbyItems []expression.Expression) (LogicalPlan, map[int]int) {
 	b.optFlag = b.optFlag | flagBuildKeyInfo
 	b.optFlag = b.optFlag | flagAggregationOptimize
+	eliminateAgg := len(gbyItems) == 0
```

Comment:
This optimize is too complex. If we have an aggregate, we set flag eliminateAgg. That's enough.